cpp/server/ct-server.cc

```diff
@@ -288,14 +292,46 @@ void WatchdogTimeout(int sig) {
   }
 }
 
+
+// Functions to handle locking for OpenSSL when used in a multithreaded env in
+// here:
+namespace {
+
+
+void locking_function(int mode, int n, const char* file, int line) {
+  static mutex* openssl_locks = new mutex[CRYPTO_num_locks()];
+  if (mode & CRYPTO_LOCK) {
+    openssl_locks[n].lock();
+  } else {
+    openssl_locks[n].unlock();
+  }
+}
+
+
+void threadid_function(CRYPTO_THREADID* id) {
+  CRYPTO_THREADID_set_numeric(id, pthread_self());
+}
+
+
+}  // namespace
+
+
 int main(int argc, char* argv[]) {
   google::ParseCommandLineFlags(&argc, &argv, true);
   google::InitGoogleLogging(argv[0]);
   google::InstallFailureSignalHandler();
+
+  // Set-up OpenSSL for multithreaded use:
+  {
```

Comment:
There was a bunch more lines in there originally, including the vector of locks which I was going to bind in, before realising this is old school and had to change it all :)
Dinged 'em.